core/editor/promotion.php

```diff
@@ -1,17 +1,48 @@
 <?php
 namespace Elementor\Core\Editor;
 
+use Elementor\Core\Utils\Promotions\Validate_Promotion;
 use Elementor\Utils;
 
 if ( ! defined( 'ABSPATH' ) ) {
 	exit; // Exit if accessed directly
 }
 
 class Promotion {
-	public function get_elements_promotion() {
-		// For BC
-		$has_pro = Utils::has_pro();
 
+	/**
+	 * @return array
+	 */
+	public function get_elements_promotion(): array {
+
+		$url = $this->get_promotion_url();
+
+		$promotion_data = $this->get_promotion_data( $url );
+
+		$new_promotion_data = apply_filters( 'elementor/editor/promotion/get_elements_promotion', $promotion_data );
+
+		if ( $new_promotion_data && count( $new_promotion_data ) <= count( $promotion_data ) ) {
+			$promotion_data = array_replace( $promotion_data, $new_promotion_data );
+			$promotion_data = $this->replace_promotion_url( $promotion_data, $url );
```

Comment:
@TzviRabinovitch 

You are already replacing the `url` in `array_replace()`, right?

If so, why do you need to do it again in `replace_promotion_url()`?

I would actually expect `replace_promotion_url()` to put back the original URL if the new one is not valid.

Reply:
@louiswol94 
that's a good point and this logic is repeated all over the promotions 
maybe we need to move it to a shared function?